From 353e98a1f9bfba0635127d4651f69a6b1bbb596d Mon Sep 17 00:00:00 2001
From: Nickolay Goppen <setotau@yandex.ru>
Date: Tue, 29 Jul 2025 00:34:48 +0300
Subject: [PATCH] arm64: dts: qcom: sdm630: add qcom,protection-domain
 properties to apr services

qcom,protection-domain property is required for adsp to properly APR functioning, such as:
 * Working q6afecc clocks
 * Opening AFE ports

Signed-off-by: Nickolay Goppen <setotau@yandex.ru>

Source-Commit: https://github.com/sdm660-mainline/linux/commit/353e98a1f9bfba0635127d4651f69a6b1bbb596d
Scope: SDM630/660-common
Porting notes: Original APR audio protection-domain properties apply unchanged; inherited by SDM660 Clover and required for q6afecc/AFE lifecycle.
---
diff --git a/arch/arm64/boot/dts/qcom/sdm630.dtsi b/arch/arm64/boot/dts/qcom/sdm630.dtsi
index fd6d1f3b341cad..9ea64cd2fe9268 100644
--- a/arch/arm64/boot/dts/qcom/sdm630.dtsi
+++ b/arch/arm64/boot/dts/qcom/sdm630.dtsi
@@ -2480,6 +2480,7 @@
 					q6afe: service@4 {
 						compatible = "qcom,q6afe";
 						reg = <APR_SVC_AFE>;
+						qcom,protection-domain = "avs/audio", "msm/adsp/audio_pd";
 
 						q6afecc: clock-controller {
 							compatible = "qcom,q6afe-clocks";
@@ -2497,6 +2498,8 @@
 					q6asm: service@7 {
 						compatible = "qcom,q6asm";
 						reg = <APR_SVC_ASM>;
+						qcom,protection-domain = "avs/audio", "msm/adsp/audio_pd";
+
 						q6asmdai: dais {
 							compatible = "qcom,q6asm-dais";
 							#address-cells = <1>;
@@ -2509,6 +2512,8 @@
 					q6adm: service@8 {
 						compatible = "qcom,q6adm";
 						reg = <APR_SVC_ADM>;
+						qcom,protection-domain = "avs/audio", "msm/adsp/audio_pd";
+
 						q6routing: routing {
 							compatible = "qcom,q6adm-routing";
 							#sound-dai-cells = <0>;

